.github/workflows/precommit.yml

```diff
@@ -1,12 +1,12 @@
 name: Gradle Precommit
 on: [pull_request]
- 
+
 jobs:
   precommit:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest] 
+        os: [windows-latest, macos-latest]  # precommit on ubuntu-latest is run as a part of the gradle-check workflow
```

Comment:
Curious if we should exclude precommit from gradle-check workflow instead (`./gradlew check -x precommit`)? Not sure what is better ....

Reply:
We are trying to exclude `precommit` only for ubuntu because gradle check runs on [ubuntu-latest](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/gradle-check.yml#L21) which includes precommit already. We will need precommit run for Windows and Mac seperately as we don't have a gradle check for them.

Reply:
Thanks @owaiskazi19, yeah I got that, I was thinking if we:
 - run precommit for all OSes
 - manually exclude precommit for Ubuntu in `gradle-check` workflow
But hopefully we will figure out how to have Gradle checks for all platforms at some point :) All good to me the way you did it.

Reply:
Ah I see. That's a good point which needs to be handled [here](https://github.com/opensearch-project/opensearch-build/blob/main/scripts/gradle/gradle-check.sh).